pkg/cmd/issue/edit/edit.go

```diff
@@ -141,75 +150,125 @@ func editRun(opts *EditOptions) error {
+		if err != nil {
+			return err
+		}
+	}
+
 	lookupFields := []string{"id", "number", "title", "body", "url"}
-	if opts.Interactive || editable.Assignees.Edited {
+	if editable.Assignees.Edited {
 		lookupFields = append(lookupFields, "assignees")
 	}
-	if opts.Interactive || editable.Labels.Edited {
+	if editable.Labels.Edited {
 		lookupFields = append(lookupFields, "labels")
 	}
-	if opts.Interactive || editable.Projects.Edited {
+	if editable.Projects.Edited {
 		lookupFields = append(lookupFields, "projectCards")
 		lookupFields = append(lookupFields, "projectItems")
 	}
-	if opts.Interactive || editable.Milestone.Edited {
+	if editable.Milestone.Edited {
 		lookupFields = append(lookupFields, "milestone")
 	}
 
-	issue, repo, err := shared.IssueFromArgWithFields(httpClient, opts.BaseRepo, opts.SelectorArg, lookupFields)
+	// Get all specified issues and make sure they are within the same repo.
+	issues, repo, err := shared.IssuesFromArgsWithFields(httpClient, opts.BaseRepo, opts.SelectorArgs, lookupFields)
 	if err != nil {
 		return err
 	}
 
-	editable.Title.Default = issue.Title
-	editable.Body.Default = issue.Body
-	editable.Assignees.Default = issue.Assignees.Logins()
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	if opts.workers == 0 {
+		opts.workers = 10
 	}
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(opts.workers)
+
+	issueURLs := make([]string, 0, len(issues))
+	issuesChan := make(chan string)
+	doneCh := make(chan any)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				doneCh <- nil
+				return
+			case issueURL, ok := <-issuesChan:
+				if ok {
+					issueURLs = append(issueURLs, issueURL)
+				}
+			}
+		}
+	}()
+
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating %d issues", len(issues)))
+	for _, issue := range issues {
+		// Copy variables to capture in the go routine below.
+		issue := *issue
+		editable := editable
```

Comment:
As the `prSahred.Editable` struct contains slices I think we are going to have to do a more in depth copy in order to not have data race conditions.